### DISCUSSION TITLE
Convert axios response to Response() object to cache API requests

### DISCUSSION THREAD

**Discussion by christianholland** [axios/axios#5857] [Q&A] [answered]
### Describe the issue

Hi there,

I would like to cache API requests using the [`cache API`](https://github.com/arthurfiorette/axios-cache-interceptor). The API response is expected to be of object [`Response()`](https://developer.mozilla.org/en-US/docs/Web/API/Response) which is not the case for the responses coming from axios. Does anyone know how I can convert an axios response to a `Response` object?

Guidance would be highly appreicated.

Thanks,
Christian

**Comment by NextThread** [comment, marked as answer]
I haven't gone through the codebase, but here's how we can convert an axis response into a Response object.

```
`
import axios from 'axios';
function convertToResponseLike(axiosResponse) {
  const { data, status, statusText, headers } = axiosResponse;

  class ResponseLike {
    constructor(responseData, init = {}) {
      this.body = responseData;
      this.status = init.status || 200;
      this.statusText = init.statusText || 'OK';
      this.headers = init.headers || new Headers();
    }

    async json() {
      return JSON.parse(this.body);
    }
  }

  return new ResponseLike(JSON.stringify(data), {
    status,
    statusText,
    headers,
  });
}
axios.get('https://api.example.com/data')
  .then((axiosResponse) => {
    const responseLike = convertToResponseLike(axiosResponse);
    caches.open('my-cache').then((cache) => {
      cache.put('cached-response', responseLike);
    });
  })
  .catch((error) => {
    console.error(error);
  });

`

```

**Comment by christianholland** [comment]
Hi @NextThread,

please apologize the very much delayed answer. Your solution worked for me, many thanks!

**Comment by NextThread** [comment]
@christianholland I'm glad, it helped :)
